### PR TITLE
Mm/game_service

### PR DESCRIPTION
**Pull request by MarchandMD**
Necessary checkmarks:

    [x] All Tests are Passing

    [x] The code will run locally

Type of change

    [x] New feature
    [] Bug Fix
    [] Refactoring

Implements/Fixes:

Adds `GameService` (with tests for happy path) and `#search` method

Check the correct boxes

    [x] This broke nothing
    [] This broke some stuff
    [] This broke everything

Testing Changes

    [x] I wrote new tests
    [] No Tests have been changed
    [] Some Tests have been changed
    [] All of the Tests have been changed(Please describe what in the world happened)
    
    Testing coverage: 100%

Checklist:

    [] My code has no unused/commented out code
    [x] I have reviewed my code
    [x] I have fully tested my code

### CODE REVIEW

**Review by DarbySmith — CHANGES_REQUESTED** [review]
CircleCI is failing due to a failing test in the service. Discussed adding VCR/Webmock for the test

**Comment by MarchandMD** [comment]
Solved issue by adding an Environment Variable on CircleCI, with the followings steps, at CircleCI website dashboard:

1. Projects (LH sidebar)
2. 3 dots for correct repo (RH side)
3. Project Settings (drop down popped up)
4. Environment variables (LH sidebar again)
5. Add Environment variables (blue button)
6. Add name and value

Done

Re-ran CI

File: spec/rails_helper.rb [hunk 1]
```diff
@@ -71,3 +71,10 @@
     with.library :rails
   end
 end
+
+VCR.configure do |config|
+  config.cassette_library_dir = "spec/fixtures/vcr_cassettes"
+  config.hook_into :webmock
+  config.filter_sensitive_data('<client_id>') { ENV.fetch('client_id') }
+  config.configure_rspec_metadata!
```
Comment (bkeener7) [inline spec/rails_helper.rb:79]:
```
config.default_cassette_options = {
    record: :new_episodes,
    re_record_interval: 7.days
  }
```

Would be a good addition eventually later. Although this configuration works fine for now, so we can potentially add this in later PR.

**Review by bkeener7 — APPROVED** [review]
Looks like CircleCI is passing now, so I will go ahead and pull this.

**Review by bkeener7 — APPROVED** [review]

**Review by DarbySmith — APPROVED** [review]
Looks good. Thanks for adding VCR and Webmock. Will merge this PR

File: spec/rails_helper.rb (hunk 1 above)
Comment (MarchandMD) [reply spec/rails_helper.rb:79]: this has been added in PR #24

**Pull request merged**